server/runtime.go

```diff
@@ -37,16 +39,117 @@ func (s *Server) Version(ctx context.Context, req *pb.VersionRequest) (*pb.Versi
 func (s *Server) CreatePodSandbox(ctx context.Context, req *pb.CreatePodSandboxRequest) (*pb.CreatePodSandboxResponse, error) {
 	var err error
 
-	// TODO: Parametrize as a global argument to ocid
-	ocidSandboxDir := "/var/lib/ocid/sandbox"
-	podSandboxDir := filepath.Join(ocidSandboxDir, req.GetConfig().GetName())
+	if err := os.MkdirAll(s.sandboxDir, 0755); err != nil {
+		return nil, err
+	}
+
+	// process req.Name
+	name := req.GetConfig().GetName()
+	if name == "" {
+		return nil, fmt.Errorf("PodSandboxConfig.Name should not be empty")
+	}
 
+	podSandboxDir := filepath.Join(s.sandboxDir, name)
+	if _, err := os.Stat(podSandboxDir); err == nil {
+		return nil, fmt.Errorf("pod sandbox (%s) already exists", podSandboxDir)
+	}
+
+	if err := os.MkdirAll(podSandboxDir, 0755); err != nil {
+		return nil, err
+	}
+
+	// creates a spec Generator with the default spec.
 	g := generate.New()
 
-	// TODO: Customize the config per the settings in the req
+	// process req.Hostname
+	hostname := req.GetConfig().GetHostname()
+	if hostname != "" {
+		g.SetHostname(hostname)
+	}
+
+	// process req.LogDirectory
+	logDir := req.GetConfig().GetLogDirectory()
+	if logDir == "" {
+		logDir = fmt.Sprintf("/var/log/ocid/pods/%s", name)
+	}
+
+	// TODO: construct /etc/resolv.conf based on dnsOpts.
+	dnsOpts := req.GetConfig().GetDnsOptions()
+	fmt.Println(dnsOpts)
+
+	// TODO: the unit of cpu here is cores. How to map it into specs.Spec.Linux.Resouces.CPU?
```

Comment:
Read this document to understand how the conversion works. Basically we need to set shares, quota and period. 
http://kubernetes.io/docs/user-guide/compute-resources/